packages/workspace/src/core/project-graph/project-graph-builder.ts

```diff
@@ -21,12 +21,16 @@ export class ProjectGraphBuilder {
     }
   }
 
-  addNode(node: ProjectGraphNode) {
+  addNode(node: ProjectGraphNode, mergeData: (d: any) => any = x => x) {
+    const prev = this.nodes[node.name];
```

Comment:
This is when a locator finds a project that's in the incremental dep graph?

Shouldn't the data be overwritten completely?

Reply:
It's a bit of leaky abstraction just for project files because we keep a cached version of it. So during construction we may not actually have a corresponding `fileMap` for the project (since its calculations are skipped).

There may be a better way to handle this.